### PR TITLE
Add salesforce Contract and Rate fields

### PR DESCRIPTION
**Pull request by macrael**
## Summary

Add a few fields for the salesforce integration:
Contract:
* lastUpdatedForDisplay
* dateContractDocsExecuted
* webURL

RateRevision:
* consolidatedRateProgramIDs

Rate: 
* webURL

#### Related issues
https://jiraent.cms.gov/browse/MCR-4609
https://jiraent.cms.gov/browse/MCR-4635

### CODE REVIEW

**Comment by macrael** [comment]
Ended up having to update a bunch of GraphQL fragment references

File: services/app-api/src/resolvers/contract/contractResolver.ts [hunk 1]
```diff
@@ -3,19 +3,53 @@ import type { Resolvers, SubmissionReason } from '../../gen/gqlServer'
...
 import { NotFoundError } from '../../postgres'
 import {
     setErrorAttributesOnActiveSpan,
     setResolverDetailsOnActiveSpan,
 } from '../attributeHelper'
 import { convertToIndexQuestionsPayload } from '../../postgres/questionResponse'
+import type { Context } from '../../handlers/apollo_gql'
 
-export function contractResolver(store: Store): Resolvers['Contract'] {
+// this is probably a little delicate type-wise. But seems worth it not to be duplicating the same resolver in two places.
```
Comment (pearl-truss) [inline services/app-api/src/resolvers/contract/contractResolver.ts:20]: Nice, I think this is a good consolidation

File: services/app-api/src/resolvers/contract/contractResolver.ts [hunk 2]
```diff
@@ -44,7 +78,31 @@ export function contractResolver(store: Store): Resolvers['Contract'] {
...
-        packageSubmissions(parent) {
+        dateContractDocsExecuted(parent: ParentType) {
+            let dateFirstSubmitted: Date | undefined
+            for (const sub of parent.packageSubmissions) {
+                if (
+                    sub.contractRevision.formData.contractExecutionStatus ===
+                    'EXECUTED'
+                ) {
+                    dateFirstSubmitted =
+                        sub.contractRevision.submitInfo?.updatedAt
```
Comment (pearl-truss) [inline services/app-api/src/resolvers/contract/contractResolver.ts:89]: I think this if statement is missing a return here. I'm seeing that a fully executed contract is having this date set to null

Screenshot 2024-11-26 at 3 13 19 PM

**Review by pearl-truss — APPROVED** [review]
I left a comment on the `dateContractDocsExecuted` but I'm seeing everything else looks good

**Pull request merged**